src/Symfony/Component/AssetMapper/Path/LocalPublicAssetsFilesystem.php

```diff
@@ -27,17 +34,32 @@ public function write(string $path, string $contents): void
         $targetPath = $this->publicDir.'/'.ltrim($path, '/');
 
         $this->filesystem->dumpFile($targetPath, $contents);
+        $this->compress($targetPath);
     }
 
     public function copy(string $originPath, string $path): void
     {
         $targetPath = $this->publicDir.'/'.ltrim($path, '/');
 
         $this->filesystem->copy($originPath, $targetPath, true);
+        $this->compress($targetPath);
     }
 
     public function getDestinationPath(): string
     {
         return $this->publicDir;
     }
+
+    private function compress($targetPath): void
+    {
+        foreach ($this->extensionsToCompress as $ext) {
+            if (!str_ends_with($targetPath, ".$ext")) {
+                continue;
+            }
+
+            $this->compressor?->compress($targetPath);
```

Comment:
Should this be checked earlier / once ? Not sure the performance cost is high though

Reply:
In reality, if `$this->extensionsToCompress`, we're (almost) sure that `$this->compressor` is not null. I left the `?` mostly to make static analyzers happy.

Reply:
In the scenario where no compressor is injected in the LocalPublicAssetsFilesystem, then extensions list will be  empty..  so the foreach works as a nulltest yep. Missed that first read sorry :)